hw1/server/src/authenticateUser.ts

```diff
@@ -0,0 +1,35 @@
+import { users } from "./assets/users";
+import { User } from "./types/UserType";
+
+class AuthenticateUser {
+
+    private username: string;
+    private password: string;
+
+    constructor(username: string, password: string) {
+
+        this.username = username;
+        this.password = password;
+
+    }
+
+   public auth(): Promise<User> {
+
+        return new Promise((res, req) => {
+
+            setTimeout(() => {
+                
+                 const foundUsers = [...users].filter(user => user.username === this.username && user.password === this.password)
+
+                 if (foundUsers.length > 0) res(foundUsers[0]);
```

Comment:
using find() let you to avoid magic numbers in code such as `foundUsers[0]` why [0] not [1] or [2] ?